neard/src/config.rs

```diff
@@ -161,8 +175,25 @@ pub struct Network {
     pub external_address: String,
     /// Comma separated list of nodes to connect to.
     pub boot_nodes: String,
-    /// Maximum number of peers.
-    pub max_peers: u32,
+    /// Maximum number of active peers. Hard limit.
+    #[serde(default = "default_max_num_peers")]
+    pub max_num_peers: u32,
+    /// Minimum outbound connections a peer should have to avoid eclipse attacks.
+    #[serde(default = "default_minimum_outbound_connections")]
+    pub minimum_outbound_peers: u32,
+    /// Lower bound of the ideal number of connections.
+    #[serde(default = "default_ideal_connections_lo")]
+    pub ideal_connections_lo: u32,
+    /// Upper bound of the ideal number of connections.
+    #[serde(default = "default_ideal_connections_hi")]
+    pub ideal_connections_hi: u32,
+    /// Peers which last message is was within this period of time are considered active recent peers (in seconds).
+    #[serde(default = "default_peer_recent_time_window")]
+    pub peer_recent_time_window: u64,
```

Comment:
Why is this not of type `Duration`?

Reply:
I guess because of serialization, just noticed that duration can be equally serialized, will update.

Reply:
@bowenwang1996 done